src/js/apps/forms/form/form_app.js

```diff
@@ -105,19 +105,22 @@ export default App.extend({
 
     this.listenTo(formService, {
       'success'(response) {
-        response.set({ _created_at: dayjs().format() });
-        this.responses.unshift(response);
-        this.setState({ responseId: response.id });
-
         const saveButtonType = this.getState('saveButtonType');
 
         if (saveButtonType === 'saveAndGoBack') {
-          Radio.request('history', 'go:back');
-          return;
+          this.listenTo(this.loadingModal, 'destroy', () => {
+            Radio.request('history', 'go:back');
+          });
         }
 
-        this.showFormStatus();
-        this.showFormUpdate();
+        if (saveButtonType !== 'saveAndGoBack') {
+          response.set({ _created_at: dayjs().format() });
+          this.responses.unshift(response);
+          this.setState({ responseId: response.id });
+
+          this.showFormStatus();
+          this.showFormUpdate();
+        }
```

Comment:
If user is using the `Save + Go Back` button, don't update the form content, status, or save button while the loading modal (showing the `Saving your work...` animation) is open.